### ISSUE TITLE
BBOK item saves in cart and should not if user doesn't check out

### ISSUE THREAD

**Issue by suehamre** [tastefully-simple/cornerstone#297]
### Expected behavior
When a user starts the join process and does not complete, we cannot have the BBOK item saving to the cart because we don't want them purchasing this item without signing up to be a consultant and cannot be included on an order with other items. 

### Actual behavior
If the user does not complete the entire join process the BBOK item is saving to their cart and they can checkout with it on a regular order that includes other items. Also, the consultant that was selected in the join process is listed in the banner. 

### Steps to reproduce behavior
Start the join process and continue through to the personal information page: 
![image](https://user-images.githubusercontent.com/57115316/91332229-7e180900-e791-11ea-8f10-ea73a8579c05.png)
Don’t checkout – BBOK is saved in users cart and the consultant that was chosen on the join path is listed in the banner. 
![image](https://user-images.githubusercontent.com/57115316/91332268-8ec87f00-e791-11ea-9aaa-a5b226f65a2f.png)
![image](https://user-images.githubusercontent.com/57115316/91332296-95ef8d00-e791-11ea-9216-b3171386602a.png)
![image](https://user-images.githubusercontent.com/57115316/91332312-9d169b00-e791-11ea-9bd6-6a7ec24e9f0b.png)
user then gets the join congratulations page: 
![image](https://user-images.githubusercontent.com/57115316/91332348-aa338a00-e791-11ea-8084-8b390b92099c.png)
Order is created in big commerce: 
![image](https://user-images.githubusercontent.com/57115316/91332390-b6b7e280-e791-11ea-9b15-67b6bc266a8c.png)
But I don't see it in the SQL table, and am not seeing any errors in the order download. 
![image](https://user-images.githubusercontent.com/57115316/91332438-cb947600-e791-11ea-8411-6a24880057d5.png)

**Comment by jsmnglng-w** [comment]
@suehamre I can't replicate the consultant that was chosen on the join path is listed in the banner. So what I did was in the personal information page, I searched and selected a sponsor
![Screen Shot 2020-08-26 at 1 30 30 PM](https://user-images.githubusercontent.com/69249028/91342696-02be5380-e7a1-11ea-8e70-f020b0bbd960.png)
And then didn't checkout, I went to the cart, and the consultant I selected from the join process was not in the banner and was not saved in the cookie. The only one persisted is the BBOK product in the cart page. 
![Screen Shot 2020-08-26 at 1 31 12 PM](https://user-images.githubusercontent.com/69249028/91342948-69437180-e7a1-11ea-9f79-2e9d0799b8b7.png)


Can you double check it? Or I guess am I replicating it correctly?

**Comment by suehamre** [comment]
@jsmnglng-w , Weird, I can't dup it now either for the consultant displaying. Just tried and got the same results you did. To confirm, yes you are replicating correctly. I even clicked on checkout just to see if that was it, but then I just got this error and still did not see the consultant in the banner when I went back in. 
![image](https://user-images.githubusercontent.com/57115316/91343836-b542e600-e7a2-11ea-8d5a-e5b9707fffbf.png)
